compiler/qsc_data_structures/src/namespaces/tests.rs

```diff
@@ -0,0 +1,365 @@
+                                                    value: NamespaceTreeNode {
+                                                        children: {},
+                                                        id: NamespaceId(
+                                                            5,
+                                                        ),
+                                                    },
+                                                },
+                                            },
+                                            id: NamespaceId(
+                                                2,
+                                            ),
+                                        },
+                                    },
+                                },
+                                id: NamespaceId(
+                                    1,
+                                ),
+                            },
+                        },
+                        "ns2": RefCell {
+                            value: NamespaceTreeNode {
+                                children: {
+                                    "nsc": RefCell {
+                                        value: NamespaceTreeNode {
+                                            children: {},
+                                            id: NamespaceId(
+                                                18,
+                                            ),
+                                        },
+                                    },
+                                    "nsb": RefCell {
+                                        value: NamespaceTreeNode {
+                                            children: {},
+                                            id: NamespaceId(
+                                                17,
+                                            ),
+                                        },
+                                    },
+                                    "nsa": RefCell {
+                                        value: NamespaceTreeNode {
+                                            children: {},
+                                            id: NamespaceId(
+                                                16,
+                                            ),
+                                        },
+                                    },
+                                },
+                                id: NamespaceId(
+                                    15,
+                                ),
+                            },
+                        },
+                    },
+                    id: NamespaceId(
+                        0,
+                    ),
+                },
+            },
+            memo: RefCell {
+                value: {},
+            },
+        }
+    "#]]
+    .assert_debug_eq(&root);
+}
+
+#[allow(clippy::too_many_lines)]
+#[test]
+fn test_find_id() {
```

Comment:
Could you modify this test to also search for the intermediate-level namespaces: "ns0", "ns1", and "ns2"?